avocado_vt/test.py

```diff
@@ -217,28 +218,48 @@ def setUp(self):
         Avocado-vt uses custom setUp/test/tearDown handling and unlike
         Avocado it allows skipping tests from any phase. To convince
         Avocado to allow skips let's say our tests run during setUp
-        phase and don't do anything during runTest/tearDown.
+        phase and report the status in test.
         """
         env_lang = os.environ.get('LANG')
         os.environ['LANG'] = 'C'
         try:
             self._runTest()
+            self.__status = "PASS"
         # This trick will give better reporting of virt tests being executed
         # into avocado (skips, warns and errors will display correctly)
-        except error.TestNAError, details:
-            raise exceptions.TestSkipError(details)
-        except error.TestWarn, details:
-            raise exceptions.TestWarn(details)
-        except error.TestError, details:
-            raise exceptions.TestError(details)
-        except error.TestFail, details:
-            raise exceptions.TestFail(details)
+        except:  # Old-style exceptions are not inherited from Exception()
+            details = sys.exc_info()[1]
+            self.__status = details
+            if not hasattr(self, "cancel"):     # Old Avocado, skip here
+                if isinstance(self.__status, error.TestNAError):
+                    if hasattr(exceptions, "TestSetupSkip"):
+                        raise exceptions.TestSetupSkip(self.__status)
+                    else:
+                        raise exceptions.TestSkipError(self.__status)
         finally:
             if env_lang:
                 os.environ['LANG'] = env_lang
             else:
                 del os.environ['LANG']
 
+    def runTest(self):
+        """
+        This only reports the results
+
+        The actual testing happens inside setUp stage, this only
+        reports the correct results
+        """
+        if self.__status == "PASS":
+            pass
+        elif isinstance(self.__status, error.TestNAError):
+            self.cancel(self.__status)
+        elif isinstance(self.__status, error.TestWarn):
+            self.log.warn(details)
+        elif isinstance(self.__status, error.TestFail):
+            self.fail(self.__status)
+        else:
+            raise self.__status
```

Comment:
Avocado-vt does not use custom handling anymore https://github.com/avocado-framework/avocado-vt/commit/37a315b8577fc8355fdd3b7c5b928f8507206eb2 so the Avocado base class will treat any unknown exceptions as TestError.